web/app/help/operations/compute-segment-expansion.asciidoc

```diff
@@ -0,0 +1,25 @@
+### Compute segment expansion
+
+Computes the expansion of each segment in a non-overlapping segmentation.
```

Comment:
Interesting that all these ops need non-overlapping segments. The definitions don't seem to need it. I guess that's for some implementation shortcuts?

Reply:
> Interesting that all these ops need non-overlapping segments. The definitions don't seem to need it. I guess that's for some implementation shortcuts?

They take a `Partition` as the input. The `Partition` is non-overlapping and the `Cover` is overlapping in NetworKit. (I think? Correct me if you think that's not how they work.)

Reply:
Oh yeah, I think you capture the networkit API correctly, my musing were more about why _they_ decided to work with a partition and not a cover and I meant it must be an implementation shortcut on _their_ end. No need to spend time to investigate anything, it is what it is, just wanted to point out that it's a bit curious.

Reply:
As I originally implemented these measures that derive from `LocalPartitionEvaluation`, I will try answering this question. First, I did not need them for overlapping communities/segments as you call them, just for non-overlapping ones.  Second, if I recall correctly, it was also not clear to me how they should be generalized to overlapping communities/segments but looking at them again now I do not see why there should be any issue at least with measures like conductance and expansion that are basically just defined for single communities/segments. As a cover stores a separate data structure for every node that contains the ids of all communities it is part of, this is quite some overhead compared to the single integer that is stored for a partition which is why converting everything to a cover might not always be a good idea. Nevertheless, I agree that it would be good to implement them for overlapping instead of disjoint communities and support disjoint communities, e.g., via some adapter. I created networkit/networkit#650 to document these ideas in NetworKit.

Reply:
Thanks for the explanation Michael, makes perfect sense! We do not have any pressing need right now to apply these on overlapping communities, but of course would be great to have what you propose in https://github.com/networkit/networkit/issues/650, thanks for creating it. So no hurry from our end, but we will certainly extend our boxes as well to overlaps if that is done.